### PR TITLE
Converting supermarket server setup into a custom resource/provider

### PR DESCRIPTION
**Pull request by shruthi-venkateswaran**

### CODE REVIEW

**Comment by irvingpop** [comment]
Thanks @shruthi-venkateswaran , this is great!

One additional thing needs to be done in order to maintain backwards compatibility and give some extra flexibility to the users:   the ability to specify additional configuration flags.

In the default recipe, I would do something like:
```ruby
supermarket_server 'supermarket' do
  chef_server_url node['supermarket_omnibus']['chef_server_url']
  chef_oauth2_app_id node['supermarket_omnibus']['chef_oauth2_app_id']
  chef_oauth2_secret node['supermarket_omnibus']['chef_oauth2_secret']
  chef_oauth2_verify_ssl node['supermarket_omnibus']['chef_oauth2_verify_ssl']
  config node['supermarket_omnibus']['config'].to_hash
end
```

where `node['supermarket_omnibus']['config']` contains additional tunables that could be added via attributes to override the [default values from the omnibus package](https://github.com/chef/omnibus-supermarket/blob/master/cookbooks/omnibus-supermarket/attributes/default.rb).

Then in the provider you could do

```ruby
def merged_supermarket_config
  new_resource.config.merge(supermarket_config)
end
```

or something similar and then pass that to the `file` resource

**Comment by irvingpop** [comment]
Great work @shruthi-venkateswaran - I am :+1: on this

File: recipes/default.rb [hunk 1]
```diff
@@ -5,65 +5,11 @@
 # Copyright (c) 2014 The Authors, All Rights Reserved.
 
 # Configure Supermarket server hostname in /etc/hosts if it isn't there (AWS)
```
Comment (robbkidd) [inline recipes/default.rb]: Maybe replace the comment here with something about how the `supermarket_server` resource below is a compatibility feature for folks still using the default recipe directly. :grin:

**Comment by robbkidd** [comment]
Nice @shruthi-venkateswaran! Resource providers for the win!

I'm :+1: aside from a little update to a comment that is no longer true.

**Comment by shruthi-venkateswaran** [comment]
@robbkidd good catch about the comment. Missed removing that. I've replaced it.

**Comment by robbkidd** [comment]
:100:

I felt bad about commenting on a comment. But now :+1: :+1:

**Pull request merged**